static/app/utils/performance/contexts/pageAlert.tsx

```diff
@@ -68,15 +76,36 @@ export function PageAlertProvider({children}: {children: React.ReactNode}) {
 
 export function PageAlert() {
   const {pageAlert} = useContext(pageErrorContext);
+  const [dismissedAlerts, setDismissedAlerts] = useLocalStorageState<string[]>(
+    localStorageKey,
+    []
+  );
+
   if (!pageAlert || !pageAlert.message) {
     return null;
   }
 
-  const isStringError = typeof pageAlert === 'string';
-  const message = isStringError ? pageAlert : pageAlert.message;
+  const {message, dismissId} = pageAlert;
+
+  if (dismissId && dismissedAlerts.includes(dismissId)) {
+    return null;
+  }
+
+  const handleDismiss = () => {
+    if (!dismissId || dismissedAlerts.includes(dismissId)) {
+      return;
+    }
+    const prev = new Set(dismissedAlerts);
+    setDismissedAlerts([...prev, dismissId]);
```

Comment:
I think the spread means that we still risk getting duplicates added, and we should only call the setter if we see there was a new `dismissId` added

```suggestion
    const newDismissedAlerts = new Set(dismissedAlerts);
    newDismissedAlerts.add(dismissId);
    
    if (newDismissedAlerts.size !== dismissedAlerts.length) {
      setDismissedAlerts([...newDismissedAlerts]);
    }
```

Reply:
I think this should be already covered with the following code we have already
```
if (!dismissId || dismissedAlerts.includes(dismissId)) {
      return;
    }
```